src/service.cc

```diff
@@ -1058,8 +1053,7 @@ int service::handle_async_check_result(check_result* queued_check_result) {
   time_t current_time = 0L;
   int state_was_logged = false;
   std::string old_plugin_output;
-  objectlist* check_servicelist = nullptr;
-  objectlist* servicelist_item = nullptr;
+  std::list<service*> check_servicelist;
```

Comment:
if it is possible you can perhaps switch to std::list<shared_ptr<service>>

Reply:
The response as for hosts